### DISCUSSION TITLE
Use DeriveEntityModel without having a Relation enum

### DISCUSSION THREAD

**Discussion by LasseRosenow** [SeaQL/sea-orm#644] [Ideas]
## Summary

Right now, when I use the `DeriveEntityModel` `Derive Macro`, I need to have a `enum Relation`, which derives `DeriveRelation`.
Otherwise I get the error: `cannot find type `Relation` in this scope`.

## Motivation

The reason why I would like to use the `DeriveEntityModel` macro without having the Relation enum is because I don't use the Relation enum.

I like to define my relations like this:

```rust

impl Related<super::homes::Entity> for Entity {
    fn to() -> RelationDef {
        Entity::belongs_to(super::homes::Entity)
            .from(Column::HomeId)
            .to(super::homes::Column::Id)
            .into()
    }
}

```

So there really is no need and it works... Right now I just have an empty Relation enum and that just feels a bit useless and adds unnecessary clutter.

**Comment by billy1624** [comment]
Hey @LasseRosenow, welcome! Take the entity below as an example, you're saying that you want to get rid of the empty `Relation` Enum completely?

```rs
use sea_orm::entity::prelude::*;
use serde::{Deserialize, Serialize};

#[derive(Clone, Debug, PartialEq, DeriveEntityModel, Deserialize, Serialize)]
#[sea_orm(table_name = "posts")]
pub struct Model {
    #[sea_orm(primary_key)]
    #[serde(skip_deserializing)]
    pub id: i32,
    pub title: String,
    #[sea_orm(column_type = "Text")]
    pub text: String,
}

#[derive(Copy, Clone, Debug, EnumIter, DeriveRelation)]
pub enum Relation {}

impl ActiveModelBehavior for ActiveModel {}
```

**Comment by LasseRosenow** [comment]
@billy1624 yes. As far as I understand, I don't really need the `Relation` enum because all relations between entities are created by using the `related<>` trait.

So I didn't like to have things at multiple places so I got rid of all the stuff inside the Relation enum and moved it inside the related traits.

So long answer short: yes I would like not to have those empty structures that I don't use, but if there are technical reasons as to why they need to exist, that would okay too of course :)

**Comment by LasseRosenow** [comment]
This all of course would not be necessary if the Relation enum derive macro would work as documented.

![Screenshot_20220323-143114_Chrome.png](https://user-images.githubusercontent.com/10547444/159711230-e1fa5fd0-6865-4dfb-9072-d90cd7261cb3.png)

See this screenshot from the documentation says that the whole first code block could be replaced by the one at the bottom which is using the macro, but I could not reproduce this.

When I only used the lower one with the macro, I still had to write the `related<>` traits as well. Is this a bug, or is the documentation wrong about that the enum with the macro replaces everything above?

Because if I could specify all my relations just with the Relation enum using macros, that would be cool as well and a good alternative to getting rid of the enum and doing everything with the `related<>` trait and there would be less boilerplate as well

**Comment by tyt2y3** [comment]
The macro annotation only handles the impl of the `RelationTrait` but not the `Related` trait.
And your usage is perfectly fine, and yes you probably don't need the empty enum.
I think in order for us to properly support that, we'd depend on https://rust-lang.github.io/rfcs/2532-associated-type-defaults.html (https://github.com/rust-lang/rust/issues/29661)
